### PR TITLE
[4.0] New Backend Template

### PR DESCRIPTION
**Pull request by wilsonge**
This is the main body of work on the redesigned backend template per @svom and @coolcat-creations awesome new design in the Joomla Magazine from November.

https://magazine.joomla.org/issues/issue-nov-2017/item/3289-episode-iv-a-new-user-interface-for-the-joomla-backend

Clearly this isn't finished but should be as useable as the current one (so yes that means it's missing mobile design, it's missing RTL etc)

This is a PR so that some basic sanity testing can happen before it is merged and so the system tests can be fixed

## TODO BEFORE MERGING
- [x] Fix system tests (probably merge https://github.com/joomla/test-system/pull/45 and then swap the default branch here back to the main 4.0-dev branch)
- [ ] Fix hound (it only errors on the compiled es5 files and a boostrap vendor file so i guess it's just setting up the right ignores)

### CODE REVIEW

**Comment by brianteeman** [comment]
I would say that as soon as the automated tests pass then this should be merged. its too big and unwieldy a pr to really comment on anything else right now. Especially as we know its not complete.

**Comment by brianteeman** [comment]
I know the menu has changed but why have the admin presets been removed? The ability to import  preset is still present but now errors

File: administrator/templates/atum/language/en-GB/en-GB.tpl_atum.ini [hunk 1]
```diff
@@ -5,12 +5,11 @@
 
 ATUM="Atum Administrator template"
 TPL_ATUM_CLEAR_CACHE="Clear Cache"
-TPL_ATUM_CONTROL_PANEL_MENU="Toggle main menu"
-TPL_ATUM_GOTO_JOOMLA_HOME_PAGE="Go to Joomla! organisation home page"
```
Comment (brianteeman) [inline administrator/templates/atum/language/en-GB/en-GB.tpl_atum.ini:9]: Both these strings are being used in the error pages

File: administrator/language/en-GB/en-GB.mod_menu.ini [hunk 2]
```diff
@@ -74,9 +75,9 @@ MOD_MENU_IMPORTANT_ITEM_MODULE_MANAGER="Module Manager"
 MOD_MENU_IMPORTANT_ITEM_COMPONENTS_CONTAINER="Components Container"
 MOD_MENU_IMPORTANT_ITEMS_INACCESSIBLE_LIST_WARNING="The administrator menu <strong>%1$s</strong> does not contain - <strong>%2$s</strong>. Select to <strong><a href='%3$s'>turn on the menu recovery mode</a></strong>."
 MOD_MENU_INSTALLER_SUBMENU_DATABASE="Database"
-MOD_MENU_INSTALLER_SUBMENU_DISCOVER="Discover"
-MOD_MENU_INSTALLER_SUBMENU_INSTALL="Install"
-MOD_MENU_INSTALLER_SUBMENU_LANGUAGES="Install Languages"
+
+
+
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.mod_menu.ini:80]: I dont understand the logic behind renaming these language constants. I thought I did understand the logic but then it hasnt been consistently applied for all the similar constants

File: administrator/templates/atum/index.php [hunk 3]
```diff
@@ -72,55 +72,48 @@
...
 		<?php if (!$hidden) : ?>
 		<div id="sidebar-wrapper" class="sidebar-wrapper" <?php echo $hidden ? 'data-hidden="' . $hidden . '"' : ''; ?>>
-			<div id="main-brand" class="main-brand align-items-center">
-				<a href="<?php echo Route::_('index.php'); ?>" aria-label="<?php echo Text::_('TPL_BACK_TO_CONTROL_PANEL'); ?>">
-					<img src="<?php echo $logoLg; ?>" class="logo" alt="<?php echo $sitename; ?>">
-				</a>
-			</div>
 			<jdoc:include type="modules" name="menu" style="none" />
+			<div id="main-brand" class="main-brand d-flex align-items-center justify-content-center">
+				<img src="<?php echo $logo; ?>" alt="<?php echo $sitename; ?>">
```
Comment (brianteeman) [inline administrator/templates/atum/index.php:100]: This is a decorative image and should have an alt =""

https://www.w3.org/WAI/tutorials/images/decorative/

File: administrator/templates/atum/language/en-GB/en-GB.tpl_atum.ini (hunk 1 above)
Comment (mbabker) [reply administrator/templates/atum/language/en-GB/en-GB.tpl_atum.ini:9]: Actually, why is this directory here at all?  None of the core extensions use local language folders...

Comment (brianteeman) [reply administrator/templates/atum/language/en-GB/en-GB.tpl_atum.ini:9]: Some of them do in j3 - mainly the templates 
Not going to get into thee discussion of why
;)

File: administrator/components/com_cpanel/View/System/HtmlView.php [hunk 4]
```diff
@@ -0,0 +1,501 @@
...
+
+			static::$notEmpty = true;
+		}
+
+		// Install
+		if ($user->authorise('core.manage', 'com_installer'))
+		{
+			// Install
+			$links['MOD_MENU_INSTALL'] = [
+				'com_installer_install' => static::arrayBuilder(
```
Comment (brianteeman) [inline administrator/components/com_cpanel/View/System/HtmlView.php:77]: shouldnt this code and the code in administrator/components/com_cpanel/View/Help/HtmlView.php be the same?

File: administrator/modules/mod_login/tmpl/default.php [hunk 5]
```diff
@@ -54,41 +67,39 @@ class="form-control input-full"
...
+						tabindex="3"
+					>
+				</div>
 			</div>
 		<?php endif; ?>
 		<?php if (!empty($langs)) : ?>
 			<div class="form-group">
-				<label for="lang" class="sr-only">
-					<?php echo Text::_('JDEFAULTLANGUAGE'); ?>
+				<label class="text-white" for="lang" class="sr-only">
```
Comment (brianteeman) [inline administrator/modules/mod_login/tmpl/default.php]: invalid markup - you can';t have two class

**Comment by mbabker** [comment]
Someone tell me it's an intentional decision to not link to the module manager ANYWHERE but the control panel quick icon please.

**Comment by coolcat-creations** [comment]
@mbabker https://projects.invisionapp.com/share/8SES9JLZB#/screens/266894724

**Comment by infograf768** [comment]
@coolcat-creations 
I do not have this layout here at all. Including the icons

I just updated an existing test site via eclipse. Anything I should do?

**Comment by coolcat-creations** [comment]
@infograf768 - as @wilsonge stated in the PR it's absolutely not in the state to be finished ...
Please take the invision prototype and the jcm article as a reference...

**Comment by dgrammatiko** [comment]
@mbabker the menu transformation is not complete atm! There are still items that need to be removed from the menu and others that need to be added (eg modules), the first step was to get the 2 views system and help in place and then patch the rest. This is not the final,  production ready code, even the menu logic (js) is not yet updated, so please bear with us as we’ll try to finish this task

**Comment by Hackwar** [comment]
It would be really good if only production-ready code would be commited to this repository... It doesn't look like this PR was ready to be merged.

**Comment by infograf768** [comment]
> Please take the invision prototype and the jcm article as a reference...

No idea what it means but I will be patient. 
Nevertheless please do add the Save and Close button separately.
We really miss that...

**Comment by dgrammatiko** [comment]
@Hackwar I think due to lack of people working on the other repo the decision was to move the code here so more devs can contribute and thus accelerate the process. The j4 menu is known to been far from production ready code since the first alpha, nothing new here

**Comment by ghost** [comment]
@infograf768 looks like this:

bildschirmfoto 2018-07-10 um 10 59 40

Source: https://projects.invisionapp.com/share/8SES9JLZB#/screens/267320995

**Comment by coolcat-creations** [comment]
@infograf768 

Link to the article: https://magazine.joomla.org/issues/issue-nov-2017/item/3289-episode-iv-a-new-user-interface-for-the-joomla-backend

Link to the prototype (Mentioned in the article): https://projects.invisionapp.com/share/8SES9JLZB#/screens/266861164_Desktop_-_Login

**Comment by Hackwar** [comment]
@dgrammatiko I don't want to complain to much, because I've been guilty of this in the past, too, but I still want us to be able to release at any time with this repo. Merging unfinished code prevents that. From a quick glance, this is in the state of work-in-progress and thus not even in alpha state. Don't get me wrong, I value the work that is being done here, but I'd rather have this split up into X PRs which are only merged when ready. Even if it looks a bit strange from time to time because new code doesn't really fit visually with old code, at least everything would work...

**Comment by brianteeman** [comment]
> but I still want us to be able to release at any time with this repo.

for reasons (that i don't agree with) it was decided that this was not to be the case at this stage of J4 development

**Comment by mbabker** [comment]
> 1.4.1 Stable master branches
> It is critical to our mission that the master branch of each major version of a Joomla product is always stable and ready for a release within a short time. This is supported by our Contributor Policy.

https://developer.joomla.org/development-strategy.html

**Comment by rdeutz** [comment]
good that we have renamed the main branch to staging :-P

**Comment by wilsonge** [comment]
I've said since JAB that I'm also starting to tighten up on the stuff I'm merging as I understand the repo now needs to move towards stability. Getting the new template design in was the one last missing part. Given the existing template was also broken in many ways I'd rather stop getting PRs to fix something that is broken and getting replaced and put in something that is nearly as 

PR for the missing module menu item: https://github.com/joomla/joomla-cms/pull/21032

**Comment by dgrammatiko** [comment]
> always stable and ready for a release within a short time

So how the heck an `alpha` or even `beta` version can be stable and ready for a release within a short time?
By definition `alpha` and `beta` are nowhere near production ready or have we also have another definition for alpha and beta in this project?

I'm confused 🤷‍♂️

**Comment by Bakual** [comment]
> So how the heck an alpha or even beta version can be stable and ready for a release within a short time?

If you look at our staging (J3) branch, it is basically production ready even when not tagged at all. "Alpha" doesn't mean it is broken (it may of course but doesn't have to). It means that the API may still change. "Beta" also doesn't mean it is broken, but it is "Feature Complete" and we don't expect any API changes anymore. RCs are then when we don't expect any changes anymore to the stable release except when we detect some new bugs, languages are usually also frozen at this point.
See https://en.wikipedia.org/wiki/Software_release_life_cycle.

**Comment by rdeutz** [comment]
4.0-dev is a development branch so the rule doesn't apply, our staging branch should be always in a state that we can make a release in a short time frame. If we have a 4.0-staging branch, then the rule will apply.

**Comment by zero-24** [comment]
@wilsonge this is a rater imported PR to be added to the 4.0 milestone :P Just for the record i have also added the new feature label.

**Pull request merged**